### ISSUE TITLE
interval of Interest retransmissions

### ISSUE THREAD

**Issue by PeterKietzmann** [cn-uofbasel/ccn-lite#225]
Most platforms (e.g.,  [ccnl-lnxkernel](https://github.com/cn-uofbasel/ccn-lite/blob/master/src/ccnl-lnxkernel/ccn-lite-lnxkernel.c#L457)) call [`ccnl_do_ageing`](https://github.com/cn-uofbasel/ccn-lite/blob/master/src/ccnl-core/src/ccnl-relay.c#L764) periodically (once a second). In that function, network layer retransmissions are handled if the max. number of retransmissions has not exceeded. Assuming the application wants to send an Interest after 0,9 seconds, just 0,1 seconds later the ageing function sends the first retransmission (in case the corresponding data hasn't arrived within that 0,1 second).
- Is this observation correct?
- Is this behavior correct?

**Comment by blacksheeep** [comment]
Sorry, I do not get the question completely.
Can you specify:
* your goal is to delay the forwarding? or
* you observing an interest that is received 0.9 seconds after the last ageing call?

in case 2 your observation is correct and the behavior may not be optimal but is intended.

**Comment by PeterKietzmann** [comment]
@blacksheeep thanks for your fast response. My goal is to observe the behavior correctly. I think the above observation lies somewhere between your two points, or I also didn't get your answer correctly. Let me rephrase: 
When I send (or forward) an Interest packet it can happen that the first retransmission appears "shortly" after the initial  packet has been sent. More specifically, this is due to the periodic `ccnl_do_ageing` call which triggers retransmissions. If the ageing function is coincidentally called 1ms after the initial packet was sent,  the first retransmission is also triggered 1ms consequently.

**Comment by blacksheeep** [comment]
Yes, this observation is correct, and the behavior is intended by the
design of the ageing function.
I am not saying it is necessary good behavior.


Peter Kietzmann <notifications@github.com> schrieb am Do., 5. Apr. 2018,
12:25:
